pkg/sql/catalog/descpb/privilege.go

```diff
@@ -266,36 +266,40 @@ func MaybeFixPrivileges(id ID, ptr **PrivilegeDescriptor) bool {
 	return modified
 }
 
-// Validate is called when writing a database, table or type descriptor.
-// It takes the descriptor ID which is used to determine if
-// it belongs to a system descriptor, in which case the maximum
-// set of allowed privileges is looked up and applied.
+// Validate returns an error if the privilege descriptor is invalid.
+// It requires the ID of the descriptor it is applied on to determine whether
+// it is is a system descriptor, because superusers do not always have full
+// privileges for those.
+// It requires the objectType to determine the superset of privileges allowed
+// for regular users.
 func (p PrivilegeDescriptor) Validate(id ID, objectType privilege.ObjectType) error {
-	allowedPrivileges := DefaultSuperuserPrivileges
+	allowedSuperuserPrivileges := DefaultSuperuserPrivileges
+	maybeSystem := ""
 
 	if IsReservedID(id) {
 		var ok bool
-		allowedPrivileges, ok = SystemAllowedPrivileges[id]
+		maybeSystem = "system "
+		allowedSuperuserPrivileges, ok = SystemAllowedPrivileges[id]
 		if !ok {
-			return fmt.Errorf("no allowed privileges found for system %s with ID=%d",
-				objectType, id)
+			return fmt.Errorf("no allowed privileges defined for %s%s with ID=%d",
+				maybeSystem, objectType, id)
 		}
 	}
 
 	// Check "root" user.
-	if err := p.validateRequiredSuperuser(id, allowedPrivileges, security.RootUserName(), objectType); err != nil {
+	if err := p.validateRequiredSuperuser(id, allowedSuperuserPrivileges, security.RootUserName(), objectType); err != nil {
 		return err
 	}
 
 	// We expect an "admin" role. Check that it has desired superuser permissions.
-	if err := p.validateRequiredSuperuser(id, allowedPrivileges, security.AdminRoleName(), objectType); err != nil {
+	if err := p.validateRequiredSuperuser(id, allowedSuperuserPrivileges, security.AdminRoleName(), objectType); err != nil {
 		return err
 	}
 
 	if p.Version >= OwnerVersion {
 		if p.Owner().Undefined() {
-			return errors.AssertionFailedf("found no owner for system %s with ID=%d",
-				objectType, id)
+			return errors.AssertionFailedf("found no owner for %s%s with ID=%d",
+				maybeSystem, objectType, id)
```

Comment:
For some reason the error messages here describe everything as "system" even when they're not system tables or anything.